situp-plugins/elasticsearch/src/test/java/com/amazon/situp/plugins/sink/elasticsearch/ElasticsearchSinkIT.java

```diff
@@ -276,4 +328,111 @@ private boolean deleteDirectory(final File directoryToBeDeleted) {
     }
     return directoryToBeDeleted.delete();
   }
+
+  public static boolean isODFE() {
+    final boolean isODFE = Optional.ofNullable(System.getProperty("odfe"))
+            .map("true"::equalsIgnoreCase).orElse(false);
+    if (isODFE) {
+      // currently only external cluster is supported for security enabled testing
+      if (!Optional.ofNullable(System.getProperty("tests.rest.cluster")).isPresent()) {
+        throw new RuntimeException("cluster url should be provided for security enabled ODFE testing");
+      }
+    }
+
+    return isODFE;
+  }
+
+  @Override
+  protected String getProtocol() {
+    return isODFE() ? "https" : "http";
+  }
+
+  @Override
+  protected RestClient buildClient(final Settings settings, final HttpHost[] hosts) throws IOException {
+    final RestClientBuilder builder = RestClient.builder(hosts);
+    if (isODFE()) {
+      configureHttpsClient(builder, settings);
+    } else {
+      configureClient(builder, settings);
+    }
+
+    builder.setStrictDeprecationMode(true);
+    return builder.build();
+  }
+
+  @SuppressWarnings("unchecked")
+  @After
+  protected void wipeAllODFEIndices() throws IOException {
+    final Response response = client().performRequest(new Request("GET", "/_cat/indices?format=json&expand_wildcards=all"));
```

Comment:
This may be really easy with the high level client

Reply:
I doublechecked the RestClient Java API. The high level client can only wraps around the RestClient.Builder(), which is not in the state variable of `ESRestTestCase`. If we try to reconstruct such a builder with proper settings. It will also be vulnerable to bugs. So taking the tradeoff into consideration I tend to keep the manual parsing logic as it is since it works for now.